pkg/authorization/service.go

```diff
@@ -107,23 +107,21 @@ func (a *AuthService) checkPermission(userID, domainID, domainType, resource, ac
 }
 
 func (a *AuthService) CreateGroup(domainID string, domainType string, groupName string, role string) error {
-	validRoles := map[string][]string{
-		DomainOrg:    {RoleOrgViewer, RoleOrgAdmin, RoleOrgOwner},
-		DomainCanvas: {RoleCanvasViewer, RoleCanvasAdmin, RoleCanvasOwner},
-	}
-
-	if roles, exists := validRoles[domainType]; exists {
-		if !contains(roles, role) {
-			return fmt.Errorf("invalid role %s for domain type %s", role, domainType)
-		}
-	} else {
+	// Validate domain type
+	if domainType != DomainOrg && domainType != DomainCanvas {
 		return fmt.Errorf("invalid domain type %s", domainType)
 	}
 
 	domain := fmt.Sprintf("%s:%s", domainType, domainID)
-	prefixedGroupName := fmt.Sprintf("group:%s", groupName)
 	prefixedRole := fmt.Sprintf("role:%s", role)
 
+	// Check if role exists in this domain using the enforcer
+	if !a.roleExistsInDomain(role, domain) {
+		return fmt.Errorf("invalid role %s for domain type %s", role, domainType)
+	}
+
+	prefixedGroupName := fmt.Sprintf("group:%s", groupName)
```

Comment:
This domain (`%s:%s`), prefixed group name (`group:%s`) and prefixed role name (`role:%s`) logic should be in a single function that is used by all these methods.